pkg/controller/controller.go

```diff
@@ -127,6 +151,16 @@ func (c *Controller) initOperatorConfig() {
 func (c *Controller) initController() {
 	c.initClients()
 	c.initOperatorConfig()
+
+	// earliest point where we can check if the namespace to watch actually exists
+	if c.opConfig.WatchedNamespace != v1.NamespaceAll {
+		_, err := c.KubeClient.Namespaces().Get(c.opConfig.WatchedNamespace, metav1.GetOptions{})
```

Comment:
I'd write the KubeClient check as 
```
if _, err := ; err != nil {
   c.logger.Fatalf(....)
}
```
The error text can be shortened to "could not find the watched namespace %q" (see https://www.postgresql.org/docs/10/static/error-style-guide.html for couldn't vs cannot).

Does it make sense to move this into a function that I mentioned above that figures the effective namespace and also checks whether it exists?